### DISCUSSION TITLE
关于过滤(filter)这块为什么只针对于第一个dataset（headDs）?

### DISCUSSION THREAD

**Discussion by RHobart** [apache/seatunnel#623] [Q&A]
`private def batchProcessing(
    sparkSession: SparkSession,
    configBuilder: ConfigBuilder,
    staticInputs: List[BaseStaticInput],
    filters: List[BaseFilter],
    outputs: List[BaseOutput]): Unit = {

    basePrepare(sparkSession, staticInputs, filters, outputs)

    // let static input register as table for later use if needed
    val headDs = registerInputTempViewWithHead(staticInputs, sparkSession)

    // when you see this ASCII logo, waterdrop is really started.
    showWaterdropAsciiLogo()

    if (staticInputs.nonEmpty) {
      var ds = headDs #这块为什么只针对第一个ds，那么其他输入源的dataset过滤不用处理了吗

      for (f <- filters) {         
        ds = filterProcess(sparkSession, f, ds)
        registerFilterTempView(f, ds)

      }
      outputs.foreach(p => {
        outputProcess(sparkSession, p, ds)
      })

      sparkSession.stop()

    } else {
      throw new ConfigRuntimeException("Input must be configured at least once.")
    }
  }`

**Comment by RickyHuo** [comment]
@RHobart 简单的数据流是1个Input，1个或多个Filter 以及1个或多个Output组成的。因为不同的Input之间的DataSet可能是异构的，所以没有办法多个Filter在多个Input的基础上进行操作。

但是的确存在多Input的使用场景，比如多数据源之间的Join或者复杂的数据处理逻辑，这类处理我们通过把Input注册为临时表来解决，可以通过Filter中指定source_table来确定到底处理的是哪个数据源。

这类配置文件可以参考：https://github.com/InterestingLab/waterdrop/blob/master/config/complex.conf.template
